fainder/execution/parallel_processing.py

```diff
@@ -193,42 +167,40 @@ def __init__(
 
             self._init_params.append((worker_id, worker_hist_paths))
 
-        # Initialize executor with fork server method to avoid copying objects in memory
         mp_context = mp.get_context("forkserver")
-        self.executor = ProcessPoolExecutor(
-            max_workers=self.num_workers,
-            mp_context=mp_context,
-        )
-
-        # Register shutdown handler
+        self.executors = [
+            ProcessPoolExecutor(max_workers=1, mp_context=mp_context, max_tasks_per_child=10000)
```

Comment:
Why do we need `max_tasks_per_child=10000` and how did you come up with the number?